powertools-e2e-tests/README.md

```diff
@@ -0,0 +1,13 @@
+## End-to-end tests
+This module is internal and meant to be used for end-to-end (E2E) testing of Lambda Powertools for Java. 
+
+__Prerequisites__: an AWS account is needed as well as a local environment able to reach this account 
+([credentials](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials.html)).
+
+To execute the E2E tests, use the following command: `mvn clean verify -Pe2e`
+
+### Under the hood
+This module leverages the following components:
+- AWS CDK to define the infrastructure and synthesize a CloudFormation template and the assets (lambda function packages)
+- The AWS S3 SDK to push the assets on S3
+- The AWS CloudFormation SDK to deploy the template
```

Comment:
Perhaps some extra context here about the surface area we have with AWS for the different bits - e.g., "we interact with the services of the modules we are testing. This means that we write and read to live cloudwatch, x-ray, [....]. Helper classes wrap the AWS APIs to fetch the data back out for assertions", or something.

I might be going a bit over the top with the commenting here, I've just trying to capture the bits I had to dig into to map the whole E2E test module out in my 🧠

Reply:
Probably I can explain a bit, for the maintainers to jump in more easily...